### ISSUE TITLE
Compute Host showing as empty

### ISSUE THREAD

**Issue by coolpalani** [wako057/openstack-salt#3]
Hi,

I setup three node controller,compute and storage node.
I ran "main state" salt commands. https://github.com/wako057/openstack-salt

After that I tried to login horizon dashboard and checked about compute host, it seems as empty.

Could you specify the order of installations ?

Should i want execute only your salt main states or "Available States" also
Heat also not installed

**Comment by coolpalani** [comment]
Fyi

Once we rebooted the controller node, its showing compute node status in horizon.

How do we fix this problem

Also couple of major component is missing in the dashboard.

Router
Floating IP

**Comment by wako057** [comment]
I am going to test on a virtual setup, i come back to you shortly

**Comment by wako057** [comment]
I've added a vagrant directory, building 4 vm, the salt master, the controller, the compute and the storage.
It need a better release but as it I made it work.

I havent go far away with the storage, didnt try the formula, but as it, with configuring the files i've build the openstack create each network, be aware that the controller need memory or you will be block due to memory and load 
In the vagrant directory, vagrant up 
once you are log in the controller:


salt '*' state.apply openstack

salt '*' state.apply database

salt '*' state.apply keystone

salt '*' state.apply glance

salt '*' state.apply nova

salt '*' state.apply neutron

salt '*' state.apply dashboard

**Comment by coolpalani** [comment]
Thank you very much for your efforts.
I will check this your new changes and post update here.

**Comment by coolpalani** [comment]
Today I found why Router isn't display in the horizon dashboard.

We have to enable following entries in /etc/openstack-dashboard/local_settings.py file

OPENSTACK_NEUTRON_NETWORK = {
    'enable_router': True,
    'enable_quotas': True,

Finally reload the webserver configurations

service apache2 reload

**Comment by wako057** [comment]
you are right, the version of the local.py it the one for the l2 management, i need to fix it

**Comment by wako057** [comment]
fixed
